godot-macros/src/docs.rs

```diff
@@ -39,52 +39,62 @@ pub fn make_inherent_impl_docs(
     constants: &[ConstDefinition],
     signals: &[SignalDefinition],
 ) -> TokenStream {
+    /// Generates TokenStream containing field definitions for documented methods and documentation blocks for constants and signals.
     fn pieces(
         functions: &[FuncDefinition],
         signals: &[SignalDefinition],
         constants: &[ConstDefinition],
-    ) -> Option<TokenStream> {
+    ) -> TokenStream {
+        let to_tagged = |s: String, tag: &str| match s.is_empty() {
+            true => s,
+            _ => format!("<{tag}>{s}</{tag}>"),
+        };
+        let signals_block = to_tagged(
+            signals
+                .iter()
+                .filter_map(make_signal_docs)
+                .collect::<String>(),
+            "signals",
+        );
+        let constants_block = to_tagged(
+            constants
+                .iter()
+                .map(|ConstDefinition { raw_constant: x }| x)
+                .filter_map(make_constant_docs)
+                .collect::<String>(),
+            "constants",
+        );
         let methods = functions
             .iter()
-            .map(make_method_docs)
-            .collect::<Option<String>>()?;
-        let signals = signals
-            .iter()
-            .map(make_signal_docs)
-            .collect::<Option<String>>()?;
-        let constants = constants
-            .iter()
-            .map(|ConstDefinition { raw_constant: x }| x)
-            .map(make_constant_docs)
-            .collect::<Option<String>>()?;
-        let field_definition = quote! {
+            .filter_map(make_method_docs)
+            .collect::<String>();
+        let methods = match methods.is_empty() {
+            true => quote! { None },
+            _ => quote! {#methods.into()},
+        };
+        quote! {
             docs: ::godot::docs::InherentImplDocs {
                 methods: #methods,
-                signals: #signals,
-                constants: #constants,
-            }.into()
-        };
-        Some(field_definition)
+                signals_block: #signals_block,
+                constants_block: #constants_block,
+            }
+        }
     }
-    pieces(functions, signals, constants).unwrap_or_else(|| quote! { docs: None })
+    pieces(functions, signals, constants)
 }
 
 pub fn make_virtual_impl_docs(vmethods: &[ImplMember]) -> TokenStream {
-    match vmethods
+    let virtual_methods = vmethods
         .iter()
         .filter_map(|x| match x {
             venial::ImplMember::AssocFunction(f) => Some(f.clone()),
             _ => None,
         })
-        .map(make_virtual_method_docs)
-        .collect::<Option<String>>()
-    {
-        Some(vmethods) => quote! {
-            virtual_method_docs: #vmethods,
-        },
-        None => quote! {
-            virtual_method_docs: ""
-        },
+        .filter_map(make_virtual_method_docs)
+        .collect::<String>();
+    match virtual_methods.is_empty() {
```

Comment:
Please empty line before, to separate complex multi-line statements.
Again, `if`.

Reply:
if'ed and multi-lined